Unity-Technologies/com.unity.multiplayer.samples.coop
# feat: auto reconnect

### Description (*)

This PR adds automatic reconnect attempts when a players loses connection. It will try to reconnect three times before giving up and sending the player back to the main menu (unless the player uses the quit button earlier).  As a workaround until a fix for [MTT-2684](https://jira.unity3d.com/browse/MTT-2684) is available in NGO pre.7, before trying to reconnect we load an empty scene. We will wait until that fix is available before cherrypicking this to develop.
### Related Pull Requests

### Issue Number(s) (*)

Fixes issue(s): [MTT-2617](https://jira.unity3d.com/browse/MTT-2617)
### Manual testing scenarios

1. Start a game with at least one client
2. On the host, pause the game until the client disconnects
3. See the client moved to the new scene with a popup message
4. Before the end of the third attempt, unpause the host
5. See the player correctly rejoins the host
### Questions or comments

There currently is an issue when testing with debug builds or in the editor. Since the host is paused, it doesn't have the time to time out the clients on its side, so when it is unpaused and the clients reconnect to it, it considers them as duplicate connections. However, in debug builds or in the editor, duplicate connections are allowed and it will be treated as a new player instead. This issue will be fixed when #488 is merged.
### Contribution checklist
